### PR TITLE
feat: Adding functional tests for Service Account 

### LINKED ISSUE #610: Google-Auth-Library-Java needs better tests
#601 fixed a problem with Auth that got released to Maven-Central and was included in the libraries-bom, which was also released to maven-central.

We should have testing to detect this.  Ideally answering the question before release to Maven-Central: does auth actually work?

### PR DESCRIPTION
**Pull request by TimurSadykov**
Adding functional tests for ServiceAccountCredentials related scenarios of AIP 4110

Fixes #610 ☕️

### CODE REVIEW

File: oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    } catch (IOException expected) {
+      assertTrue(expected.getMessage().contains(expectedMessage));
+    }
+  }
+
+  private HttpResponse executeRequestWithCredentialsWithoutScope(String serviceUrl)
+      throws IOException {
+    final GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+    // = GoogleCredentials.getApplicationDefault();
+    // .createScoped("https://www.googleapis.com/auth/cloud-platform");
```
Comment (arithmetic1728) [inline oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java:172]: remove?

File: oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+          serviceUri, "https://www.googleapis.com/auth/adexchange.buyer");
+      fail(String.format("Should throw exception '%s'", expectedMessage));
+    } catch (IOException expected) {
+      assertTrue(expected.getMessage().contains(expectedMessage));
+    }
+  }
+
+  private HttpResponse executeRequestWithCredentialsWithoutScope(String serviceUrl)
+      throws IOException {
+    final GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
```
Comment (arithmetic1728) [inline oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java:170]: What type of credential is it? Does Java run these functional tests in kokoro?

**Review by arithmetic1728 — APPROVED** [review]

File: oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  }
+
+  private void executeRequestWrongScope(String serviceUri)
+      throws FileNotFoundException, IOException {
+    String expectedMessage = "403 Forbidden";
+
+    try {
+      executeRequestWithCredentialsWithScope(
+          serviceUri, "https://www.googleapis.com/auth/adexchange.buyer");
+      fail(String.format("Should throw exception '%s'", expectedMessage));
```
Comment (elharo) [inline oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java:162]: just concatenate here. It's simpler.

**Review by elharo — APPROVED** [review]

File: oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java (hunk 2 above)
Comment (TimurSadykov) [reply oauth2_http/javatests/com/google/auth/oauth2/functional/FTServiceAccountCredentialsTest.java:170]: it is serviceAccountCredential. Yes, it is run in Kokoro like integration tests. There is some common setup for yoshi java projects.

**Pull request merged**